File: frontend/types/analysis.ts
export type MatchScore = number;

export interface AIFeedback {
  summary: string;
  tips: string[];
  generatedAt: string;
  source: string;
}

export interface Analysis {
  id: string;
  userId: string;
  resumeId: string;
  resumeFileName: string;
  jobDescription: string;
  extractedRequirements: string[];
  matchScore: MatchScore;
  matchedRequirements: string[];
  missingRequirements: string[];
  improvementSuggestions: string[];
  aiFeedback?: AIFeedback | null;
  isSeededDemoData?: boolean;
  createdByDemoSeed?: boolean;
  isDemoCreated?: boolean;
  createdAt: string;
  updatedAt: string;
}

export interface CreateAnalysisInput {
  resumeId: string;
  jobDescription: string;
}

export interface AnalysisSummary {
  id: string;
  resumeId: string;
  resumeFileName: string;
  matchScore: MatchScore;
  matchedRequirements: string[];
  missingRequirements: string[];
  isSeededDemoData?: boolean;
  createdByDemoSeed?: boolean;
  isDemoCreated?: boolean;
  createdAt: string;
  updatedAt: string;
}

File: backend/app/schemas/analysis.py
from pydantic import BaseModel, Field


class CreateAnalysisRequest(BaseModel):
    resumeId: str = Field(min_length=1)
    jobDescription: str = Field(min_length=1)


class AnalysisResponse(BaseModel):
    id: str
    userId: str
    resumeId: str
    jobDescription: str
    matchScore: float = Field(ge=0, le=100)
    matchedSkills: list[str]
    missingSkills: list[str]
    improvementSuggestions: list[str]
    createdAt: str
    updatedAt: str


class AnalysisSummaryResponse(BaseModel):
    id: str
    resumeId: str
    matchScore: float = Field(ge=0, le=100)
    matchedSkills: list[str]
    missingSkills: list[str]
    createdAt: str
    updatedAt: str
